### DISCUSSION TITLE
"Data Views" menu

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5623] [Feature Requests]
<img width="368" alt="image" src="https://user-images.githubusercontent.com/37256050/186798021-e8d63d01-a07d-4caa-a28c-b22d91026e10.png">

> We are proposing a new menu item called Data Views or similar which offers lists of certain data elements in a scrollable way like the output of an unrestricted query. This would remove the need to have a suite of standard simple queries for a combination of some Interactions and Data Entry items like loans, borrows, permits, agents etc., and would make access to the data faster (fewer clicks). If possible, each view could have a small number of additional fields to provide further information about the data (customizable via Preferences). (Screenshots A + B)

**Requested By:** **[CSIRO](https://app.asana.com/0/1201972579963460/1202844554923815/f)**

**Comment by maxpatiiuk** [comment]
The user request is a bit ambiguous. Here is our best interpretation, with features requested by other users added on top:
Opening the queries list prompts to select a base table.
Once selected, it only shows queries for that base table.
The list also includes shared queries, read-only institution-wide queries, autogenerated query and a default query.
User can designate any one query as a default.
There is a button next to each table name on the base table selection screen which serves as a shortcut to the default query.

**Comment by CorinnaPaeper** [comment]
Hi Max. What we were trying to achieve is a reduced number of clicks to get from the landing page to a list view of objects whilst also reducing the number of ‘simple’ queries in the Queries section. We were envisioning a simple table similar to what we’d be seeing if we were to go to one of the backend tables directly.

From a workflow point-of-view, it’d be something like:

•    Click1 - open Data Views menu > scroll to table of interest (small list of 'main' tables)
•    Click2 - open table to scroll through records within table
•    Click3 - open specific item of table in form view if required.

The only way to achieve something similar currently is to create a simple query for each object type but this clutters up the queries list and still requires you to view the query details before running it.

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/diplay-all-collection-object-records/791/2

**Comment by grantfitzsimmons** [comment]
Peter from Geneva has requested a similar functionality. This is parts of both this and https://github.com/specify/specify7/issues/256.

> Many of our users wish to see all the existing records of collection objects  when they start working so that they can go to the last ones and control or update them. Of course these records can be obtained by a query, but for many users this is too complicated (actually the queries are not understood by many users who do not know what a relational database is).
So it would be good to have in Record Sets a kind of dynamic Record Set button that will show all existing records for the Collection Object form.

[@Schuchert on the Speciforum](https://discourse.specifysoftware.org/t/diplay-all-collection-object-records/791)
